lms/djangoapps/student_dashboard/tests.py

```diff
@@ -0,0 +1,36 @@
+import ddt
+import mock
+import factory
+
+from django.db.models import signals
+from django.test import TestCase, RequestFactory
+
+from common.lib.nodebb_client.categories import ForumCategory
+from lms.djangoapps.onboarding.tests.factories import UserFactory
+from lms.djangoapps.student_dashboard.views import get_joined_communities, get_recommended_xmodule_courses, \
+    get_enrolled_past_courses
+
+
+@ddt.ddt
+class TestStudenDashboard(TestCase):
+
+    @factory.django.mute_signals(signals.pre_save, signals.post_save)
+    def setUp(self):
+        self.user = UserFactory()
+        self.request = RequestFactory()
+        self.request.user = self.user
+
+    @ddt.data(200, 500)
+    def test_get_joined_communities(self, status):
+        with mock.patch.object(ForumCategory, 'joined') as joined:
+            joined.return_value = status, []
+            communities = get_joined_communities('some-username')
+            self.assertEqual(communities, [])
+
+    def test_get_recommended_xmodule_courses(self):
```

Comment:
For `get_recommended_xmodule_courses` we should write at-least one unit test with data, like we should create some courses one with a tag `test` and we should created a user profile with interests `test` and then we should assert if this method is returning the course with tag `test`.